compiler/src/main/kotlin/visitors/typechecker.kt

```diff
@@ -324,7 +325,22 @@ class TypeChecker(symbolTable: Table) : ScopedASTVisitor(symbolTable = symbolTab
 
     override fun visit(node: Identifier) {
         // Get type of name
-        node.setType(symbolTableSession.getSymbolType(node.spelling)!!)
+        val decl = symbolTableSession.getSymbol(node.spelling)!!
+        if (decl is FuncDecl) {
+            node.setType(UndeterminedType)
+            ErrorLogger += TypeError(node.ctx, "Found function name '${node.spelling}', but the function is not called.")
+        } else {
+            node.setType(
+                when (decl) {
+                    is StateDecl -> StateType
+                    is NeighbourhoodDecl -> LocalNeighbourhoodType
+                    is AssignStmt -> decl.expr.getType()
+                    is ConstDecl -> decl.expr.getType()
+                    is FunctionArgument -> decl.type
+                    else -> throw AssertionError()
```

Comment:
Why no error message for this?

Reply:
It should never happen. Guess I can give it an error message anyway, so if we change something, we get a nice warning.

Reply:
It should never happen. Guess I can give it an error message anyway, so if we change something, we get a nice warning.